### PR TITLE
Added: xbmcbin support in translatepath

### PR DESCRIPTION
**Pull request by MarcelRoozekrans**
### Functional description

The inputstream,.adaptive addon uses the special path `special://xbmcbin`

### Reasoning

Adding this is making it work. 

### Technical description

- Combined some logic in to geneic local method
- Added logic to parse the xbmcbin path and return the respected path.

### CODE REVIEW

File: xbmc.py [hunk 1]
```diff
@@ -350,23 +350,27 @@ def translatePath(path):  # NOSONAR
 
     """
 
+    def get_return_path(base_path, name, *segments):
```
Comment (basrieter) [inline xbmc.py:352]: I usually don't like nested methods, but I understand the issue in `xbmc.py` as it is just a bunch of methods anyways. 

Could you include `.replace("/", os.sep)` in the `*[i for i in segments if i and i != '']`? That would make something like `*[i.replace("/", os.sep) for i in segments if i and i != '']`

And remove it from line 365, 369 and 373?

**Review by basrieter — CHANGES_REQUESTED** [review]

File: xbmc.py (hunk 1 above)
Comment (MarcelRoozekrans) [reply xbmc.py:352]: Me neither but thought the nested methods are the pyhtonic way and thought that reusability is very low here justifying the nested method. 

Made the changes as requested so replacing all path seperators on the segments.

Comment (basrieter) [reply xbmc.py:352]: Oh, the nested method here made sense. There was no need to move it. Is it too much to ask you to move it back? 🙊 

BTW: the rest look fine.

Comment (MarcelRoozekrans) [reply xbmc.py:352]: Sure moved back

**Review by basrieter — APPROVED** [review]

**Comment by basrieter** [comment]
Thanks @MarcelRoozekrans

**Pull request merged**